client/my-sites/hosting/php-version-card/index.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import CardHeading from 'components/card-heading';
+import MaterialIcon from 'components/material-icon';
+import Button from 'components/button';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getHttpData, requestHttpData } from 'state/data-layer/http-data';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import Spinner from 'components/spinner';
+import FormRadiosBar from 'components/forms/form-radios-bar';
+import { errorNotice, successNotice } from 'state/notices/actions';
+
+const requestId = ( siteId, method ) => `hosting-php-version-${ method }-${ siteId }`;
+
+export const requestPhpVersion = siteId => {
+	const method = 'GET';
+
+	return requestHttpData(
+		requestId( siteId, method ),
+		http(
+			{
+				method,
+				path: `/sites/${ siteId }/hosting/php-version`,
+				apiNamespace: 'wpcom/v2',
+				body: {},
+			},
+			{}
+		),
+		{
+			fromApi: () => version => {
+				return [ [ requestId( siteId, method ), version ] ];
+			},
+			freshness: 0,
+		}
+	);
+};
+
+export const setPhpVersion = ( siteId, version ) => {
+	const method = 'POST';
+
+	return requestHttpData(
+		requestId( siteId, method ),
+		http(
+			{
+				method,
+				path: `/sites/${ siteId }/hosting/php-version`,
+				apiNamespace: 'wpcom/v2',
+				body: {
+					version: version,
+				},
+			},
+			{}
+		),
+		{
+			fromApi: () => success => {
+				if ( true === success ) {
+					requestPhpVersion( siteId );
+				}
+				return [ [ requestId( siteId, method ), success ] ];
+			},
+			freshness: 0,
+		}
+	);
+};
+
+class PhpVersionCard extends React.Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			selectedPhpVersion: null,
+		};
+	}
+
+	componentDidMount() {
+		requestPhpVersion( this.props.siteId );
+	}
+
+	changePhpVersion = event => {
+		const version = event.target.value;
+
+		this.setState( {
+			selectedPhpVersion: version,
+		} );
+	};
+
+	componentDidUpdate( prevProps ) {
+		const prevResponse = prevProps.updateResult;
+		const { updateResult, translate } = this.props;
+
+		const updateNoticeId = 'hosting-php-version';
+
+		if ( prevResponse && prevResponse !== 'failure' && updateResult === 'failure' ) {
+			this.props.errorNotice( translate( 'Failed to set PHP version.' ), {
```

Comment:
Optional and can be followed up on as an enhancement, but since we do want side effects, adding /php-versions apis to the data-layer may make more sense.

See https://github.com/Automattic/wp-calypso/pull/37430 as an example, though we're happy to help if you had questions

Reply:
Feels like soooo much boiler plate for something that probably ain't going to be resued. Can do tho. Learning is learning.

Reply:
After a rebase, I think what's here is fine to land. We can incrementally refactor after if you're interested in learning about this.